### PR TITLE
Issue #24587: Invoice Margin calculated incorrectly for foreign currency...

### PR DESCRIPTION
**Pull request by andersdd**
... invoices

Correct cost calculation on calcinvoiceamt function to factor in alternate currencies on the invoice.

### CODE REVIEW

**Comment by sindhupriyaav** [comment]
Performed UAT: Observed that Issue is fixed in 'Unposted Invoices' screen and still exists in 'Invoice' screen. Please find the screenshot attached for more information
margincalwrongininvoicescreen-24587

**Comment by andersdd** [comment]
Confirming margin issue on the invoice item screen.  This is an independent issue from that reported here and requires a change to the qt-client - not the xtuple repository - so this PR can go ahead and I will resolve the issue in a qt-client patch

File: foundation-database/public/functions/calcinvoiceamt.sql [hunk 1]
```diff
@@ -30,12 +30,14 @@ BEGIN
   SELECT COALESCE(SUM(ROUND((invcitem_billed * invcitem_qty_invuomratio) *
                             (invcitem_price / COALESCE(invcitem_price_invuomratio, 1.0)), 2)), 0.0),
          COALESCE(SUM(ROUND((invcitem_billed * invcitem_qty_invuomratio) *
-                            (COALESCE(coitem_unitcost, itemCost(itemsite_id), 0.0)
+                            (currtolocal(invchead_curr_id, COALESCE(coitem_unitcost, itemCost(itemsite_id), 0.0), invchead_invcdate)
```
Comment (gilmoskowitz) [inline foundation-database/public/functions/calcinvoiceamt.sql]: The sales order might not be in the base currency so this should probably be a call to `currToCurr()`.

Comment (gilmoskowitz) [inline foundation-database/public/functions/calcinvoiceamt.sql]: Never mind... I misread the field from the `coitem` table.

**Pull request merged**